include/sys/p4wq.h

```diff
@@ -30,6 +30,7 @@ struct k_p4wq_work {
 	int32_t priority;
 	int32_t deadline;
 	k_p4wq_handler_t handler;
+	void *priv;
```

Comment:
One word isn't really much to complain about, but this is a paradigm collision.  There's no reason the application code can't put exactly the same pointer into their own struct.

Not worth a -1, but I don't like this and we should only have one API style for how to find the "owner" of a work item.  Other data structures in Zephyr don't do this.